pkg/mssqlcontainer/ingest/mechanism/restore.go

```diff
@@ -0,0 +1,85 @@
+package mechanism
+
+import (
+	"fmt"
+	"github.com/microsoft/go-sqlcmd/internal/container"
+)
+
+type restore struct {
+}
+
+func (m *restore) Initialize(controller *container.Controller) {
+}
+
+func (m *restore) CopyToLocation() string {
+	return "/var/opt/mssql/backup"
+}
+
+func (m *restore) Name() string {
+	return "restore"
+}
+
+func (m *restore) FileTypes() []string {
+	return []string{"bak"}
+}
+
+func (m *restore) BringOnline(databaseName string, _ string, query func(string), options BringOnlineOptions) {
+	if options.Filename == "" {
+		panic("Filename is required for restore")
+	}
+	if databaseName == "" {
+		panic("databaseName is required for restore")
+	}
+
+	query(fmt.Sprintf(
+		m.restoreStatement(),
+		m.CopyToLocation(),
+		options.Filename,
+		databaseName,
+		m.CopyToLocation(),
+		options.Filename,
+	))
+}
+
+func (m *restore) restoreStatement() string {
```

Comment:
>oreStatement() st [](http://example.com/codeflow?start=22&length=17)

this code is a good argument for porting SMO to AoT, as SMO already has all the backup and restore logic which has been tested for years.